community/cypher/cypher-logical-plans-3.4/src/main/scala/org/neo4j/cypher/internal/v3_4/logical/plans/Eager.scala

```diff
@@ -22,7 +22,7 @@ package org.neo4j.cypher.internal.v3_4.logical.plans
 import org.neo4j.cypher.internal.util.v3_4.attribution.IdGen
 
 /**
-  * Consumes and buffers all source rows, and then produces them.
+  * Consumes and buffers all source rows, marks the transaction as stable, and then produces all rows.
```

Comment:
Out of curiosity, why do we need to do this?

Reply:
Well.

We will use the Eager to ensure that all writes from a write clause are executed before continuing the execution. This done so that we can guarantee that they are visible in later reads. However, with the two-layer tx-state they will per definition not be visible unless we also move these writes into the stable state, so that's why we need to call `markAsStable` here.